### DISCUSSION TITLE
How To

### DISCUSSION THREAD

**Discussion by lbdyck** [zopencommunity/meta#178] [General]
Having a How To document would be helpful.

Things like:

1. How to install a single port, with pre/co-reqs without zopen (e.g. git)
2. How to install an upgrade of an installed port without zopen (e.g. git)
3. How to configure the users .profile for testing of the updates
4. How to configure the system /etc/profile for production use
5. How to install (1) into a unique filesystem for testing, then migrate to a production filesystem
6. (5) using unique ZFS with mount points
7. How to get help with ZOT 
8. Where to find community help/support

**Comment by MikeFultonDev** [comment]
Great questions @lbdyck . Here are my initial thoughts:

1) When you say 'install' do you want to do this by downloading the pax file to a system, FTP the file up, then analyze the runtime dependencies to see what other files need to be uploaded, etc? This would help customers with an 'air gap' issue where they may not have connectivity to github.com from their z/OS system. Alternately, we could add support so that someone could provide an alternate server for the pax files and we could enhance _zopen_ tools to be able to connect to that system instead? 
2) same questions as #1. 
3) For this question, would it be reasonable to have a 'test' directory and then a 'prod' directory that could be used? What I'm thinking is if we have a way to let people install the software packages under test location A, then we could provide a 'zopen promote' type of tool to promote the packages to the (shared) target location after testing? The cleanest way I've seen this done is with symbolic links - the sysprog installs the new software but does not update the symbolic links for the 'prod' usage until fully tested, at which point, they switch it over.  If we take this approach, then the .profile would be changed by the sysprog/tester to point to the newly installed locations, putting them ahead of the 'prod' locations. They can be tested, and then when ok, update the links and revert the .profile.
4) Once we know where we want to put everything for shared usage, then this should be pretty simple (#110)
5) I think this ties into 1 and 3 correct? 
6) same as 5
7) I did a reasonably significant overhaul on the docs: https://zosopentools.github.io/meta/#/ Tell me if this helps. I'd also like us to add docs to the actual 'zopen' tool itself for the detailed questions
8) I added in links to the community and discussions from the main page. Suggestions welcome

**Reply by lbdyck** [reply]
My attempt at an answer:

1. Install by 'downloading' a pax file to the system, either from GitHub or by 'uploading' from another location that has access to GitHub. Not sure you need to analyze what other files are missing, other than to provide some guidance on pre/co-reqs.
2. See (1)
3. The installer (person) should be the one designating the installation locations. symlinks work fine to a point but for the z/OS system and deployment across multiple lpars, and perhaps across multiple sysplexes, the 'test' location would be promoted (copied) into a new/unique filesystem on a new/unique ZFS that would then be mounted accordingly.  For example, with the Rocket installation, a target location is defined where it is tested. Then that location (directory) is copied into a filesystem mounted on a new ZFS mounted on a ~/tmp/ point and then unmounted from there. It is then mounted on the production mount point that is referenced by /etc/profile. The ZFS is also copied to other sysplexes that don't share dasd with the installation system. The ZFS typically has a version or other qualifier.
4. ok
5. ok
6. ok
7. See https://zosopentools.github.io/meta/#/Guides/using - need to discuss how to install zopen (chicken vs egg)
See https://zosopentools.github.io/meta/#/Guides/QuickStart - 404 ( i will look more later)
8.nice

**Comment by MikeFultonDev** [comment]
@lbdyck would love to get your current view on your (excellent) 7 points. We have made progress but in particular, #3 is not something I've done enough trials with, but should be close. @DevonianTeuchter can you provide some colour commentary?

**Comment by DevonianTeuchter** [comment]
Lots to digest :-)

Installing paxes outwith the zopen ecosystem can be done much as was done before, download/expand the pax and source the .env (which will source the .app whereas zopen will just source .app); updated versions can be downloaded/expanded to a different dir and the .env sourced for that release.
There is work afoot on an updated zopen-install that deals with pax files directly, allowing installion from a custom location/repo so no reliance on GH connectivity (like in an airgapped environment).

There is the zopen init --clone command that should take an existing (eg. Test) zopen environment and clone it to a new location; the <zopenroot>/etc/zopen-config updates to reflect the new environment location. As such, a sysadmin could test in one location/mountpoint (or anyone could test if they sourced the "test" zopen-config) and when happy, --clone that to the prod environment - users could then just source /etc/zopen-config from the prod image.  Or if sufficiently confident in the environment, sysadmin could add a line to /etc/profile to source <zopenroot>/etc/zopen-config for all or users can themselves modify  $HOME/.profile|$HOME/.bash_profile

**Reply by MikeFultonDev** [reply]
I need to try the —clone. Thanks for the info

**Reply by DevonianTeuchter** [reply]
Would be good to get some feedback as not sure anyone has tried it in anger yet and there's been a fair bit of change lately...
